src/board/mod.rs

```diff
@@ -333,20 +333,27 @@ impl Board {
     //#[inline(never)] //turn off for profiling
     // Note: Same as get(), the board is indexed starting at 1-1
     pub fn play(&mut self, m: Move) -> Result<(), IllegalMove> {
-        match self.is_legal(m) {
-            Err(e) => return Err(e),
-            Ok(_)  => {}
-        }
+        try!(self.is_legal(m));
```

Comment:
Does `try!` really return the `Err`? I thought that it would just panic. That wouldn't be very nice, I think.